src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,994 @@
+                    static_cast< const Layer3D& >( geomodel_.geological_entity(
+                        GeologicalEntityType( "Layer" ), layer.index() ) );
+                builder_.geology.set_geological_entity_name(
+                    layer, unit->getInterpretedFeature()->getTitle() );
+                unit_2_info_[unit].name_ =
+                    unit->getInterpretedFeature()->getTitle();
+            }
+        }
+
+        for( auto contactIndex : range( rank_interp->getContactCount() ) )
+        {
+            unit_2_info_[rank_interp->getSubjectOfContact( contactIndex )]
+                .relation_base_ = get_contact_mode(
+                rank_interp->getSubjectContactModeOfContact( contactIndex ) );
+            unit_2_info_[rank_interp->getSubjectOfContact( contactIndex )]
+                .interface_base_ = interp_2_geo_entity_
+                [(AbstractFeatureInterpretation*) rank_interp
+                        ->getHorizonInterpretationOfContact( contactIndex )];
+
+            unit_2_info_[rank_interp->getDirectObjectOfContact( contactIndex )]
+                .relation_top_ = get_contact_mode(
+                rank_interp->getDirectObjectContactModeOfContact(
+                    contactIndex ) );
+            unit_2_info_[rank_interp->getDirectObjectOfContact( contactIndex )]
+                .interface_top_ = interp_2_geo_entity_
+                [(AbstractFeatureInterpretation*) rank_interp
+                        ->getHorizonInterpretationOfContact( contactIndex )];
+        }
+    }
+
+    void GeoModelBuilderRESQMLImpl::read_strati_column( EpcDocument& pck )
+    {
+        const std::vector< RESQML2_0_1_NS::StratigraphicColumn* >
+            stratiColumnSet = pck.getStratigraphicColumnSet();
+
+        if( stratiColumnSet.size() > 1 )
+        {
+            Logger::warn( "I/O", "RESQML2 load supports zero or one "
+                                 "stratigraphic column for the moment." );
+            return;
+        }
+
+        for( auto column : stratiColumnSet )
+        {
+            if( column->getStratigraphicColumnRankInterpretationSet().size()
+                != 1 )
+            {
+                Logger::warn( "I/O", "RESQML2 load only rank one "
+                                     "interpretation for the moment." );
+                continue;
+            }
+
+            showAllMetadata( column );
+
+            for( auto rank_interp :
+                column->getStratigraphicColumnRankInterpretationSet() )
+            {
+                if( !rank_interp->isAChronoStratiRank() )
+                {
+                    Logger::warn( "I/O",
+                        "RESQML2 load supports only chronological "
+                        "stratigraphic rank for the moment." );
+                    continue;
+                }
+
+                read_rank_interpreation( rank_interp );
+
+                // assign regions to layers
+                // TODO: this is a workaround until the
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)